### ISSUE TITLE
Could I create a "log out of all devices" button?

### ISSUE THREAD

**Issue by MaxPleaner** [atmos/sinatra_auth_github#35]
I'm trying implement a "log out of all devices" button.

Apologies for the somewhat verbose explanation that I'm about to give, but it should make the situation more apparent:

With my website, a user has a different session token for each of their devices. These session tokens get authenticated by becoming mapped to a Github username (obtained using this gem's flow).

Say the user logs in on two devices (laptop and phone), and then from the laptop clicks "log out from all devices". It's easy to log out from the device that called the request - I just put `logout!` in the controller. But how can I have the mobile device also log out? Even if the mobile client forgets about the session token, they might not have to enter their credentials/password on Github again because that's cached in the session. 

To summarize, is there any way I can store Github connection objects in a custom object and then close some arbitrary subset of them? The goal would be to force those devices to re-authenticate the username/password in Github.

**Comment by atmos** [comment]
I implemented this functionality for GitHub and I don't think it's ever going to be available to the public. The short answer is that it requires some work on the OAuth provider side to provide an endpoint where a user can verify that an OAauth client token can be tied back to a valid browser session. "Log out on all devices" would log the user out of github.com and clients would identify that their session on github.com was invalid and invalidate the cookie in the client app.

I think most people looking for this functionality end up using things like okta.

I don't use or actively maintain this project anymore so you're probably better off using something from omniauth.

**Comment by MaxPleaner** [comment]
Appreciate that. FWIW this project still works fine and is quite easy to use
